lib/src/builder_implementation.dart

```diff
@@ -4251,28 +4341,39 @@ String _formatAsConstList(String typeName, Iterable parts) =>
 
 String _formatAsDynamicList(Iterable parts) => "[${parts.join(", ")}]";
 
+String _formatAsDynamicSet(Iterable parts) => "{${parts.join(", ")}}";
+
 String _formatAsMap(Iterable parts) => "{${parts.join(", ")}}";
 
 /// Returns a [String] containing code that will evaluate to the same
 /// value when evaluated in the generated file as the given [expression]
 /// would evaluate to in [originatingLibrary].
-String _extractConstantCode(
+Future<String> _extractConstantCode(
     Expression expression,
     _ImportCollector importCollector,
     AssetId generatedLibraryId,
-    Resolver resolver) {
+    Resolver resolver) async {
   String typeAnnotationHelper(TypeAnnotation typeName) {
     LibraryElement library = typeName.type.element.library;
     String prefix = importCollector._getPrefix(library);
     return "$prefix$typeName";
   }
 
-  String helper(Expression expression) {
+  Future<String> helper(Expression expression) async {
     if (expression is ListLiteral) {
-      Iterable<String> elements =
-          expression.elements.map((Expression subExpression) {
-        return helper(subExpression);
-      });
+      List<String> elements = [];
+      for (CollectionElement collectionElement in expression.elements) {
+        if (collectionElement is Expression) {
+          Expression subExpression = collectionElement;
+          elements.add(await helper(subExpression));
+        } else {
+          // TODO(eernst) implement: `if` and `spread` elements of list.
```

Comment:
Perhaps create an issue to track this

Reply:
That one has been created already (#181).